x-pack/plugins/session_view/server/routes/process_events_route.ts

```diff
@@ -0,0 +1,101 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema } from '@kbn/config-schema';
+import type { ElasticsearchClient, Logger } from 'kibana/server';
+import { IRouter } from '../../../../../src/core/server';
+import { PROCESS_EVENTS_ROUTE, PROCESS_EVENTS_PER_PAGE } from '../../common/constants';
+import { expandDottedObject } from '../../common/utils/expand_dotted_object';
+
+export const registerProcessEventsRoute = (router: IRouter, logger: Logger) => {
+  router.get(
+    {
+      path: PROCESS_EVENTS_ROUTE,
+      validate: {
+        query: schema.object({
+          sessionEntityId: schema.string(),
+          cursor: schema.maybe(schema.string()),
+          forward: schema.maybe(schema.boolean()),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const client = context.core.elasticsearch.client.asCurrentUser;
+      const { sessionEntityId, cursor, forward = true } = request.query;
+      const body = await doSearch(client, sessionEntityId, cursor, forward);
+
+      return response.ok({ body });
+    }
+  );
+};
+
+export const doSearch = async (
+  client: ElasticsearchClient,
+  sessionEntityId: string,
+  cursor: string | undefined,
+  forward = true
+) => {
+  // Temporary hack. Updates .siem-signals-default index to include a mapping for process.entry_leader.entity_id
+  // TODO: find out how to do proper index mapping migrations...
+  let siemSignalsExists = true;
+
+  try {
+    await client.indices.putMapping({
+      index: '.siem-signals-default',
+      body: {
+        properties: {
+          'process.entry_leader.entity_id': {
+            type: 'keyword',
+          },
+        },
+      },
+    });
+  } catch (err) {
+    siemSignalsExists = false;
+  }
+
+  const indices = ['logs-endpoint.events.process-default'];
+
+  if (siemSignalsExists) {
+    indices.push('.siem-signals-default');
+  }
+
+  const search = await client.search({
+    index: indices,
+    body: {
+      query: {
+        match: {
+          'process.entry_leader.entity_id': sessionEntityId,
+        },
+      },
+      // This runtime_mappings is a temporary fix, so we are able to Query these ECS fields while they are not available
+      // TODO: Remove the runtime_mappings once process.entry_leader.entity_id is implemented to ECS
+      runtime_mappings: {
+        'process.entry_leader.entity_id': {
+          type: 'keyword',
+        },
+      },
+      size: PROCESS_EVENTS_PER_PAGE,
+      sort: [{ '@timestamp': forward ? 'asc' : 'desc' }],
+      search_after: cursor ? [cursor] : undefined,
+    },
+  });
+
+  const events = search.hits.hits.map((hit: any) => {
+    // the .siem-signals-default index flattens many properties. this util unflattens them.
+    hit._source = expandDottedObject(hit._source);
```

Comment:
Just curious as to why the need for the unflattened version? See the other comment regarding the `fields` api

Reply:
when pulling events from the .siem-signals-default index, the original document fields were flattened. I'm not sure if this is because mappings don't exist for the new fields (ECS data model changes are not merged yet). That being said, we plan to break out alerts into it's own route, so at the very least the route to load process events won't need this util function. This work will happen in a separate PR. Will def check out the fields api.